File: mobile/lib/api/events_api.dart
import '../models/event.dart';
import 'api_client.dart';

class EventsApi {
  final _client = ApiClient.instance;

  Future<List<Event>> list() async {
    final data = await _client.get('/events') as List<dynamic>;
    return data
        .map((e) => Event.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<List<Event>> upcoming() async {
    final data = await _client.get('/events/upcoming') as List<dynamic>;
    return data
        .map((e) => Event.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<void> rsvp(String eventId, String status, {String? note}) async {
    await _client.post('/events/$eventId/rsvp', data: {
      'status': status,
      if (note != null) 'note': note,
    });
  }
}

File: frontend/src/api/events.api.ts
import type { Event, EventRsvpEntry, RsvpStatus } from '../types';
import api from './client';

export const eventsApi = {
  list: (params?: {
    eventType?: string;
    from?: string;
    to?: string;
    includeCancelled?: string;
  }): Promise<Event[]> => api.get('/events', { params }),

  upcoming: (): Promise<Event[]> => api.get('/events/upcoming'),

  get: (id: string): Promise<Event> => api.get(`/events/${id}`),

  create: (data: Partial<Event>): Promise<Event> => api.post('/events', data),

  update: (id: string, data: Partial<Event>): Promise<Event> =>
    api.patch(`/events/${id}`, data),

  cancel: (id: string): Promise<Event> => api.patch(`/events/${id}/cancel`),

  rsvp: (id: string, status: RsvpStatus, note?: string): Promise<unknown> =>
    api.post(`/events/${id}/rsvp`, { status, note }),

  rsvps: (id: string): Promise<EventRsvpEntry[]> =>
    api.get(`/events/${id}/rsvps`),

  markAttendance: (
    id: string,
    entries: { userId: string; present: boolean }[],
  ): Promise<unknown> => api.post(`/events/${id}/attendance`, { entries }),
};
